src/ModAttachr.ts

```diff
@@ -48,12 +55,14 @@ export class ModAttachr implements IModAttachr {
             for (const mod of settings.mods) {
                 this.addMod(mod);
             }
-        }
+        }
+
+        this.eventNames = settings.eventNames || new EventNames();

```

Comment:
Should be the first line in the constructor. The constructor calls `this.addMod` for some mods. That can call to `this.enableMod`, which needs `this.eventNames` to exist.